GiniVision/Classes/AnalysisContainerViewController.swift

```diff
@@ -98,9 +104,29 @@ extension AnalysisContainerViewController: AnalysisDelegate {
     
     func displayError(withMessage message: String?, andAction action: NoticeAction?) {
         let notice = NoticeView(text: message ?? "", noticeType: .error, action: action)
-        DispatchQueue.main.async { 
+        DispatchQueue.main.async {
             self.showNotice(notice)
         }
     }
     
+    func displayNoResultsScreen(completion: ((_ shown: Bool) -> ())) {
```

Comment:
Why are you using a completion closure when you always call it synchronously before the method returns? Just return a boolean.

Reply:
Because clients could still use the current navigation controller, and this way they can concatenate the same animation. I really don't like this, but it is the best way (IMO) for now.

Reply:
concatenate the same animation? I don't understand.

Reply:
I was wrong. From that point they can still use the same UINavigationController, so I can just a return a boolean.